### ISSUE TITLE
false positive errors

### ISSUE THREAD

**Issue by nsajko** [tecosaur/CheckDoc.jl#6]
Reproducer:

```julia
"""
    ExampleModule

An example module.
"""
baremodule ExampleModule
    baremodule Implementation
        export name
        function name end
    end

    export public_name

    """
        public_name

    Call to do stuff.
    """
    const public_name = Implementation.name
end

using CheckDoc: checkdocs
checkdocs(ExampleModule)
```

Output:

```none
 ◀▶ Report on 3 entities in Main.ExampleModule ━ found 2 possible issues
════════════════════════════════════════════════════════════════════════
╭─● Error
│ ┌ VariablesHaveDoc: Should be documented
│ └ 1. Implementation @ unknown (within nothing)
│ ┌ PublicHaveDoc: Should be documented
╰ └ 2. Implementation.name @ unknown (within nothing)
```

The `checkdocs` function says that:
* `Implementation` "should be documented", even though it's an internal implementation detail
* `Implementation.name` "should be documented", even though it's an implementation detail (because `Implementation` isn't public)

Also, the `@ unknown (within nothing)` parts in the output are confusing and suspicious.

**Comment by tecosaur** [comment]
Thanks for opening this!

I've adjusted the source location printing so it should be a bit clearer.

Regarding asking to document internals, that's deliberate: I think internal functions should have docstrings too. That said, there are lower "levels" of compliance that you can test for, e.g.

```
julia> checkdocs(ExampleModule, CheckDoc.LEVEL1)
 ◀▶ Report on 3 entities in Main.ExampleModule ━ found 1 possible issue
═══════════════════════════════════════════════════════════════════════
╭─● Error (1)
│ ┌ PublicHaveDoc (1): Should be documented
╰ └ 1. Implementation.name @ unknown
```

Now I just need to make aliasing like this be picked up properly...

**Comment by tecosaur** [comment]
An extra complication:

```julia-repl
julia> Base.ispublic(ExampleModule.Implementation, :name)
true
```

**Comment by nsajko** [comment]
`name` is public within `Implementation`, but it's not public within `ExampleModule`.

**Comment by tecosaur** [comment]
You say that looking at the code, but I've got to determine that with the `Base.ispublic` API :stuck_out_tongue:

I think I've managed to work this out in a43610d57379bb0f5cc24d7e193ed430aedfa88d though.
